src/app/3d/qgs3dviewsmanager.h

```diff
@@ -0,0 +1,52 @@
+/***************************************************************************
+  qgs3dviewsmanager.h
+  --------------------------------------
+  Date                 : December 2021
+  Copyright            : (C) 2021 by Belgacem Nedjima
+  Email                : gb underscore nedjima at esi dot dz
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGS3DVIEWSMANAGER_H
+#define QGS3DVIEWSMANAGER_H
+
+#include "ui_qgs3dviewsmanager.h"
+
+#include <QDialog>
+#include <QStringListModel>
+#include <QDomElement>
+
+class Qgs3DMapCanvasDockWidget;
+
+class Qgs3DViewsManager : public QDialog, private Ui::Qgs3DViewsManager
+{
+    Q_OBJECT
+
+  public:
+    explicit Qgs3DViewsManager( QWidget *parent = nullptr, Qt::WindowFlags f = Qt::WindowFlags() );
+
+    void reload();
+
+    void set3DMapViewsDom( QMap<QString, QDomElement> &mapViews3DDom );
+    void set3DMapViewsWidgets( QMap<QString, Qgs3DMapCanvasDockWidget *> &mapViews3DWidgets );
+  private slots:
+    void openClicked();
+    void duplicateClicked();
+    void removeClicked();
+    void renameClicked();
+  private:
+    QStringListModel mListModel;
```

Comment:
This should instead be a pointer, with the QStringListModel parented to the dialog